### DISCUSSION TITLE
Istio Support

### DISCUSSION THREAD

**Discussion by SaschaKuhnigk** [dekorateio/dekorate#970] [Ideas]
Hello,

are there any plans to support istio? (e.g. render a VirtualService definition instead of Ingress)

**Comment by iocanel** [comment]
@SaschaKuhnigk if we can have a solid use case, I wouldn't mind adding istio support.

**Reply by SaschaKuhnigk** [reply]
Hello @iocanel,

define traffic routing by using a VirtualService instead of Ingres  might be a requirement in environments using Istio as a service mesh. 

Furthermore, it would be nice if we can utilize Istio features by using dekorate, e.g.:
- setting the istio gateway - when the service isn`t exposed by the default gateway
- define circuit breaker configuration for the service
- set traffic policies by using a DestinationRule (e.g. connection pool size, ...)
- ...
